lib/Doctrine/DBAL/Driver/SQLAnywhere/SQLAnywhereConnection.php

```diff
@@ -48,9 +48,16 @@ class SQLAnywhereConnection implements Connection, ServerInfoAwareConnection
      */
     public function __construct($dsn, $persistent = false)
     {
+        $allowedResourceTypes = array(
+            'SQLAnywhere connection'
+            ,'SQLAnywhere persistent connection ' //note the blank!
+        );
+
         $this->connection = $persistent ? @sasql_pconnect($dsn) : @sasql_connect($dsn);
 
-        if ( ! is_resource($this->connection) || get_resource_type($this->connection) !== 'SQLAnywhere connection') {
+        $resourceType = get_resource_type($this->connection);
+
+        if ( ! is_resource($this->connection) || !in_array($resourceType, $allowedResourceTypes, true)) {
```

Comment:
TBH I think we might safely remove the resource type check at all. We should just leave the `is_resource()` check, everything else is driver specific implementation detail.